techlibs/xilinx/brams.txt

```diff
@@ -90,13 +96,13 @@ match $__XILINX_RAMB18_SDP
   or_next_if_better
 endmatch
 
-#match $__XILINX_RAMB36_TDP
-#  min bits 4096
-#  min efficiency 5
-#  shuffle_enable B
-#  make_transp
-#  or_next_if_better
-#endmatch
+match $__XILINX_RAMB36_TDP
+  min bits 4096
```

Comment:
Should RAMB36 and RAMB18 have the same min bits?

Reply:
I'm not sure how this value is calculated. Why shouldn't there be less than 4k bits used, if it is otherwise the best match?

Reply:
It would be nice to assign a weight or polynomial, or some soft limit, rather than just hard limits.

Reply:
I guess that is what "efficiency" is for?

Reply:
Yeah, but it's just the percentage of used bits. For example, may some types of memory use less power or have less delay, or maybe it's better to fill one RAM 90% than use two at 95% efficiency, or one at 100% and one at 50% for a greedy algorithm. I guess you can prioritize them by changing the order, but then they would have to have the same exact efficiency for it to matter.

Anyway, I don't know where the 4k bit minimum comes from, so I didn't modify it.

Reply:
Another thing, the 8/16/32 bit modes are weird in that you can waste e.g. 1/9th of the memory and yet have an efficiency of 100% because it's 100% of the available capacity in that mode.

Reply:
> Another thing, the 8/16/32 bit modes are weird in that you can waste e.g. 1/9th of the memory and yet have an efficiency of 100% because it's 100% of the available capacity in that mode.

A narrower mode is deeper